### ISSUE TITLE
Next release discussion

### ISSUE THREAD

**Issue by bsmiles32** [mupen64plus/mupen64plus-core#107]
@richard42 mentioned that he wanted to do release very soon (maybe April or May ?). So I'd like to discuss the agenda about that.

* What needs to be done before the release (code, wiki, documentation ...) ?
* What should be postponed until after that release (which PR) ?

Not really to pressure anybody with a schedule, just to allow sharing and organizing the work between interested developers.

**Comment by Narann** [comment]
What version it would be? 2.1? I say this so we could use the _Milestone_ feature of github for tickets. :)

**Comment by richard42** [comment]
I was thinking of tagging this as version 2.5, because it's been so long since the last release, and with all the core refactoring changes it might not be long before we have to go to 3.0 anyway.

As far as what needs to be done, I would mostly like to see only bug fixes being merged (no refactoring which doesn't fix anything) and lots of testing.  When I get ready to do the actual tag and release, I'll probably make "smoke test" builds under win32/osx/linux to make sure everything builds and seems to run okay.

**Comment by tony971** [comment]
It looks like 2.5 is going to be a "because we haven't released in a long time" release. Wouldn't it be more meaningful if it signaled phase 2 of the refactorings, where the new APIs are in place but still optional? It would draw a lot of attention From front end debs. They would be signaled to update for phase 3 and version 3.0.

**Comment by richard42** [comment]
Also I want to name the next release as 2.5 because I think 2.1 and 2.2 are too similar to 2.0; they are easily mistaken for each other.  This might not be universal; it might just be me :)

**Comment by bsmiles32** [comment]
> "because we haven't released in a long time" release

That may be true to some extent, but I went through the list of commits and since last public release (3 july 2013) we got some very good stuff which have user-visible impacts (non exhaustive list) :
[CORE]
- PIF challenge fix (Banjo Tooie)
- Timing fixes related to CounterPerOp setting (various games)
- EEPROM fixes (Donkey Kong saves)
- Perfect Dark freeze fix
- Notable performance improvement for the pure interpreter

[RSP]
- JPEG task dispatching fix (Bottom of the 9th)
- Musyx support in HLE (many games)
- POLEF audio command implementation (fix PilotWings intro)
- RESAMPLE_ZOH audio command implementation (fix 1080 audio test)
- RESAMPLE_GE audio command implementation (fix GoldenEye "missing" sounds)
- NAUDIO_14 audio command implementation (fix Conker Bad fur day sound effects)

Plus many non-user visible impact aka major refactorings and synchronization with m64p-ae.
But the best thing that happened since the last release is the constitution of a team of skilled and dedicated contributors.
So yeah I think we can bump the version number to 2.5 easily, we earned it :)

**Comment by Narann** [comment]
I don't know who is responsive or the next release build. I don't know if there is a release tool chain or anything but I would humbly ask if it should be possible to compile my [fix_gles branch](https://github.com/Narann/mupen64plus-video-rice/tree/fix_gles) on all platform for me? _Like_ a real release.

I say this because as you maybe know, I removed the _pgl_ prefix and it _could_ create problem during cross plateform compilation. I know my branch build on Raspbian (Debian for RPi) and Linux x64 (my station) but I would like to at least _point_ on which config my branch fail. I don't ask for a real release, just pass through every compiler. If everything goes smooth and we can have a rice build with my changes on every plateforms (even for experimental purpose) I would love to see users test it.

The _pgl_ prefix is a problem on OpenGLES code (my fix_gles simply remove every _pgl_ prefix in favor of _gl_). If I rice could do thing right that would be perfect.

Thanks in advance!

**Comment by richard42** [comment]
I can make builds under OSX and Win32/MSVC2013 for you.  Do you want me to test the plugin as well, or just make sure this branch builds?  I don't think I have any 32-bit linux systems to test this platform.

**Comment by Narann** [comment]
Thanks Richard!

If it build and you can test it would be awesome of course! I don't want to waste your time.

I just think if it build without trouble and work properly we could finally merge it with upstream.

**Comment by richard42** [comment]
Narann, I got your rice video branch to build under Win32 by making some changes:

http://pastebin.com/29S3ickn

I tried to run it, but got a crash:

![m64p-narann-fix-gles-branch-rice-video-crash](https://cloud.githubusercontent.com/assets/6228255/7107598/9ff8d1b4-e11e-11e4-9fc6-6aba1ed8b5bb.png)

**Comment by Narann** [comment]
Thanks a lot Richard!

From the pastebin you sent, I noticed you didn't use the [fix_gles](https://github.com/Narann/mupen64plus-video-rice/tree/fix_gles) branch but the [master](https://github.com/Narann/mupen64plus-video-rice/tree/master) one.

fix_gles does not rely on Rice extension function pointer but pure SDL (with `#define GL_GLEXT_PROTOTYPES`). It would have been interesting to test.

Anyway, I think I pointed your problem and just fixed in my master.

Hope it's fine!

PS: Notice I didn't add your pastebin modifications in both master and fix_gles branches yet. I will do this asap.

**Comment by richard42** [comment]
Narann,  that is strange because after I cloned your rice video fork, I did a 'git branch' to list the branches and see if I needed to switch to fix_gles.  But it only listed the 'master' branch; there was no other.  Did I clone the wrong repo?  Or is there something special I need to do to tell it to get all of the branches?

**Comment by conchurnavid** [comment]
git branch -r

**Comment by conchurnavid** [comment]
or when you want all branches then

    git branch -a

**Comment by Narann** [comment]
Hi @richard42, does @conchurnavid command line (thanks!) fix your problem?

**Comment by richard42** [comment]
I think it's more complicated than that, but I did a little research and learned about git's behavior with respect to branch checkout.  By default it only retrieves the master branch from a remote repo when you do a clone.  There are several different options that you have for getting a local copy of another branch, depending upon what you intend to do with it.  Are you ready for me to test the fix_gles branch on win32 and osx?

**Comment by Narann** [comment]
Yes if you don't mind. :)

The commit to use should be https://github.com/Narann/mupen64plus-video-rice/commit/6597ee565631aedde2dffd798182585b42a03489 (The latest of fix_gles branch actually).

**Comment by richard42** [comment]
OK, I checked out this branch.  It doesn't compile in MSVC2013 due to the dynamic local array definitions.  Here is a fix:

http://pastebin.com/nwRTG3K8

Then the linker fails, because it cannot link against the GL functions:

http://pastebin.com/03WcVRDx

**Comment by Narann** [comment]
Thanks Richard! I'm digging into this GL extension stuff right now and I think I start to get the point of all those function pointers. Documenting myself [I realized](https://www.opengl.org/discussion_boards/showthread.php/170419-SDL-1-3-and-GLEW) there is no safe cross plateform way to load OpenGL function pointers without rely on external deps or do some hacks.

However, I think the way Rice was doing things originally was not the most cross platform one (because of the _pgl_ prefix). I've just [commit](https://github.com/Narann/mupen64plus-video-rice/commit/62f6e7b504d05ebd72067c6e738bad55d42fdc8c) a redo of the function pointer.

I've asked @Nebuleon to confirm if this work on real GLES hardware.

Roughly, I define prototypes depending on the platform W32? GLES? etc...  I didn't tried with OSX yet so any compiler errors about missing symbols would help me to categorize function declaration depending on platforms.

Anyone is welcome to test!

Also, if anyone could send me the gl.h of OSX and gl.h of Windows. I could use them to find which OS need which function.

**Comment by richard42** [comment]
I can do a build later, but OSX will have everything in OpenGL 2.1

**Comment by Nebuleon** [comment]
Narann/mupen64plus-video-rice@62f6e7b504d05ebd72067c6e738bad55d42fdc8c still works on a Linux GLES2 device. Now, that commit's purpose to build on Windows, so that will need tested too.

**Comment by Narann** [comment]
Nebuleon just confirm me my commit is working on real GLES hardware.

At the current state, OSX function pointer gathering is supposed to be the same than Linux one.

**Comment by littleguy77** [comment]
@Narann - Sorry I've been out of the loop, things have been hectic here.  I just tested your `fix_gles` branch on Android and am able to build without issue :)  I'll describe the runtime behavior at https://github.com/mupen64plus/mupen64plus-video-rice/pull/31

**Comment by hyperresonance** [comment]
What's the status on this? I am planning a banjo kazooie tas and would like to get started on it, and i've been told since August this is being worked on with no real updates.

**Comment by Narann** [comment]
Hello!
If by "on this" you mean the release, [it's out](https://github.com/mupen64plus/mupen64plus-core/releases/tag/2.5)!

**Comment by rlabrecque** [comment]
Lets close this, and create a fresh one to plan out the next release?
